src/components/v2/appDetails/k8Resource/K8Resource.component.tsx

```diff
@@ -41,12 +48,85 @@ export const K8ResourceComponent = ({
     isDeploymentBlocked,
     isExternalApp,
 }: K8ResourceComponentProps) => {
+    const history = useHistory()
+    const location = useLocation()
+    const currentNode = useParams<{ nodeType: string }>().nodeType
+    const currentFilter = useSearchString().searchParams.filterType || ALL_RESOURCE_KIND_FILTER
     const [nodes] = useSharedState(IndexStore.getAppDetailsNodes(), IndexStore.getAppDetailsNodesObservable())
     const { isSuperAdmin } = useMainContext()
     useEffect(() => {
         AppDetailsStore.markAppDetailsTabActiveByIdentifier(AppDetailsTabs.k8s_Resources)
     }, [])
 
+    useEffect(() => {
+        IndexStore.updateFilterType(currentFilter.toUpperCase())
+    }, [currentFilter, nodes])
+
+    // nodes according to current filter
+    const currentFilteredNodes = useMemo(
+        () =>
+            nodes.filter(
+                (node) => currentFilter === ALL_RESOURCE_KIND_FILTER || doesNodeSatisfiesFilter(node, currentFilter),
+            ),
+        [currentFilter, nodes],
+    )
```

Comment:
verify if this re-computed on each render. I noticed we added `${nodes}` somewhere

Reply:
not computed on every render, we are getting nodes here from IndexStore